clients/roscpp/include/ros/service.h

```diff
@@ -58,40 +58,79 @@ namespace service
  * @param service_name The name of the service.
  * @param req The request message.
  * @param[out] res Storage for the response message.
+ * @param timeout The amount of time to wait for, in milliseconds.  If timeout is -1,
```

Comment:
Just added `ros::service::call` top-level versions with `timeout`.

I copied the doxygen documentation from other functions in the file. That's why it says `milliseconds`. I'm going to fix that together with the changes suggested by @flixr 

I've also added a unit test, which uses this new function (as the other unit test in the same test suite).

Reply:
Done